dinh_ngoc_do/laravel-app/database/migrations/2021_09_03_130458_create_products_table.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateProductsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('products', function (Blueprint $table) {
+            $table->id();
+            $table->string('name');
+            $table->integer('price');
+            $table->integer('quantity');
+            $table->date('start_date');
+            $table->text('description');
```

Comment:
xem field nào ko required thì thêm `->nullable()` vào nhé

Reply:
![5](https://user-images.githubusercontent.com/83223275/132135155-4365f3a7-6c59-494c-95e7-c8065a64b86c.png)
anh ơi em bổ sung rồi nhưng mà hàm down thì e chưa biết viết như nào vì e chỉ chỉnh trạng thái các column null từ no -> yes thôi , vì em xem doc thì k có modifier nào để chỉnh lại từ null -> not null cả

Reply:
cái trong hàm down kia là em copy trên xuống trước cho dễ sửa thôi

Reply:
File này up là create thì down là drop là chuẩn r

Reply:
còn change từ null -> not null thì như này ` $table->integer('quantity');`